### PR TITLE
Update lang and distribution versions

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/3128
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec

### CODE REVIEW

File: ballerina/Ballerina.toml [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 keywords = ["level", "format"]
 repository = "https://github.com/ballerina-platform/module-ballerina-log"
 icon = "icon.png"
 license = ["Apache-2.0"]
-distribution = "2201.0.4"
+distribution = "2201.2.0"
 
 [[platform.java11.dependency]]
-path = "../native/build/libs/log-native-2.3.0.jar"
+path = "../native/build/libs/log-native-2.4.0-SNAPSHOT.jar"
```
Comment (daneshk) [inline ballerina/Ballerina.toml:13]: Can we add maven config as well here

File: ballerina/Ballerina.toml [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 
 [[platform.java11.dependency]]
-path = "../test-utils/build/libs/log-test-utils-2.3.0.jar"
+path = "../test-utils/build/libs/log-test-utils-2.4.0-SNAPSHOT.jar"
```
Comment (daneshk) [inline ballerina/Ballerina.toml:16]: This should be in `testonly` scope

File: ballerina/Ballerina.toml (hunk 1 above)
Comment (kalaiyarasiganeshalingam) [reply ballerina/Ballerina.toml:16]: Updated now.

File: ballerina/Ballerina.toml (hunk 2 above)
Comment (kalaiyarasiganeshalingam) [reply ballerina/Ballerina.toml:19]: Updated now.

File: integration-tests/Ballerina.toml [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-path = "../native/build/libs/log-native-@project.version@.jar"
+path = "../native/build/libs/log-native-2.4.0-SNAPSHOT.jar"
 
 [[platform.java11.dependency]]
-path = "../test-utils/build/libs/log-test-utils-@project.version@.jar"
+path = "../test-utils/build/libs/log-test-utils-2.4.0-SNAPSHOT.jar"
 
 [[platform.java11.dependency]]
-path = "./lib/io-native-@io.native.version@.jar"
+path = "./lib/io-native-1.3.0-20220718-152300-9eb8217.jar"
```
[the remark this answers is not in the document]
Comment (kalaiyarasiganeshalingam) [reply integration-tests/Ballerina.toml]: This file changes when running the test. After the test succeeds, these changes are cleaned. 

The test failed locally. So, this was added in the PR. So I reverted it.

File: gradle.properties [hunk 4]
```diff
@@ -14,6 +14,6 @@ ballerinaLangVersion=2201.2.0-20220719-103500-236cc180
 #stdlib dependencies
 stdlibIoVersion=1.3.0-20220718-152300-9eb8217
 stdlibRegexVersion=1.3.0
-observeVersion=1.0.4
+observeVersion=1.0.5
 observeInternalVersion=1.0.3
```
Comment (niveathika) [inline gradle.properties:18]:
```suggestion
observeInternalVersion=1.0.4
```

**Review by niveathika — APPROVED** [review]

**Pull request merged**